From d4a1dd479b242f8e82ba6d4a5836ceb6ea274764 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 12:58:44 -0400
Subject: [ethernet] qlge: remove open-coded skb_cow_head

Message-id: <1408625927-3701-14-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88530
O-Subject: [PATCH RHEL 7.1 13/16] qlge: remove open-coded skb_cow_head.
Bugzilla: 1089357
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: francois romieu <romieu@fr.zoreil.com>

Bug 1089357 (https://bugzilla.redhat.com/show_bug.cgi?id=1089357)

net commit id bb9689e631a6e5bb8544773168a05fcd49828d08

Signed-off-by: Francois Romieu <romieu@fr.zoreil.com>
Cc: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
Cc: Shahed Shaikh <shahed.shaikh@qlogic.com>
Cc: Ron Mercer <ron.mercer@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlge/qlge_main.c b/drivers/net/ethernet/qlogic/qlge/qlge_main.c
index 2d8d05d..91f161d 100644
--- a/drivers/net/ethernet/qlogic/qlge/qlge_main.c
+++ b/drivers/net/ethernet/qlogic/qlge/qlge_main.c
@@ -2558,11 +2558,9 @@ static int ql_tso(struct sk_buff *skb, struct ob_mac_tso_iocb_req *mac_iocb_ptr)
   int err;
   __be16 l3_proto = vlan_get_protocol(skb);
 
-  if (skb_header_cloned(skb)) {
-   err = pskb_expand_head(skb, 0, 0, GFP_ATOMIC);
-   if (err)
-    return err;
-  }
+  err = skb_cow_head(skb, 0);
+  if (err < 0)
+   return err;
 
   mac_iocb_ptr->opcode = OPCODE_OB_MAC_TSO_IOCB;
   mac_iocb_ptr->flags3 |= OB_MAC_TSO_IOCB_IC;
-- 
1.7.1